internal/commands/app/create_test.go

```diff
@@ -335,75 +330,78 @@ Check out your app: cd ./test-app && realm-cli app describe
 		}
 
-		clientZipPkg, err := zip.OpenReader("testdata/react-native.zip")
+		defaultBackendZipPkg, err := zip.OpenReader("testdata/project.zip")
 		assert.Nil(t, err)
-		rc.ClientTemplateFn = func(groupID, appID, templateID string) (*zip.Reader, bool, error) {
-			return &clientZipPkg.Reader, true, nil
-		}
-		ac := mock.AtlasClient{}
-		ac.GroupsFn = func() ([]atlas.Group, error) {
-			return []atlas.Group{{ID: "123"}}, nil
+
+		realmClient := mock.RealmClient{
+			FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+				return []realm.App{}, nil
+			},
+			ExportFn: func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
+				return "", &defaultBackendZipPkg.Reader, nil
+			},
+			CreateAppFn: func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+				return realm.App{
+					GroupID:     groupID,
+					ID:          "456",
+					ClientAppID: name + "-abcde",
+					Name:        name,
+					AppMeta:     meta,
+				}, nil
+			},
+			ImportFn: func(groupID, appID string, appData interface{}) error {
+				return nil
+			},
 		}
-		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
-			return []atlas.Cluster{{Name: "test-cluster"}}, nil
+		atlasClient := mock.AtlasClient{
+			ClustersFn: func(groupID string) ([]atlas.Cluster, error) {
+				return []atlas.Cluster{{Name: "test-cluster"}}, nil
+			},
 		}
 
 		cmd := &CommandCreate{createInputs{
 			newAppInputs: newAppInputs{
-				Name:            "template-app",
-				Location:        realm.LocationVirginia,
+				Name:            "bitcoin-miner",
+				Project:         testApp.GroupID,
+				Location:        realm.LocationIreland,
 				DeploymentModel: realm.DeploymentModelGlobal,
-				ConfigVersion:   realm.DefaultAppConfigVersion,
 			},
 			Clusters: []string{"test-cluster"},
 		}}
 
-		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: rc, Atlas: ac}))
+		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: realmClient, Atlas: atlasClient}))
 
-		console.Tty().Close() // flush the writers
-		<-doneCh              // wait for procedure to complete
+		_, err = local.LoadApp(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name))
+		assert.Nil(t, err)
 
-		path := filepath.Join(profile.WorkingDirectory, cmd.inputs.Name, local.BackendPath)
-		appLocal, err := local.LoadApp(path)
+		backendFileInfo, err := ioutil.ReadDir(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name))
 		assert.Nil(t, err)
 
-		assert.Equal(t, appLocal.RootDir, path)
+		// we expect for the command to have created a default app. we will assert that realm_config.json exists and that
+		// backend/ and frontend/ do not exist in the directory.
+		var containsRealmConfig bool
+		var containsTemplateDirChild bool
+		for _, fi := range backendFileInfo {
+			if strings.Contains(fi.Name(), local.NameRealmConfig) {
+				containsRealmConfig = true
+			}
+			if fi.Name() == local.BackendPath || fi.Name() == local.FrontendPath {
```

Comment:
is it worth just using the same style of `strings.Contains` checks for these?

Reply:
Hmm, I think I'd rather make the realm_config.json be an exact match than keeping the `strings.Contains` just so the assertion is as strict as possible.

I'll do that now but if you feel differently, I can change it

Reply:
the exact match on `realm_config.json` is fine by me since ultimately the assert here is that `false` went to `true` (b/c this condition hit)

what worries me is the opposite style of assert for making sure we don't have `backend` or `frontend` directories.  if for w/e reason the `fi.Name()` started having a trailing slash or any other characters in it, this test wouldn't catch that (b/c `false` would have remained `false`).  that's why I'm wondering if it makes sense to at least use the more permissive `strings.Contains` check for this frontend/backend logic.

[question] instead of using `ioutil.ReadDir` on the whole project directory and iterating over its files, why don't we just explicitly look for `realm_config.json` (& verify it exists) and `backend`/`frontend` directories (& verify they don't exist)?

Reply:
oh i see what you mean. That's true.

I think your suggestion here is good. I'll go ahead and do that